File: platform/android/MapboxGLAndroidSDK/src/main/java/org/maplibre/android/location/MapLibreFloatAnimator.java
package org.maplibre.android.location;

import android.animation.FloatEvaluator;
import android.animation.TypeEvaluator;

import androidx.annotation.NonNull;
import androidx.annotation.Size;

class MapLibreFloatAnimator extends MapLibreAnimator<Float> {
  MapLibreFloatAnimator(@NonNull @Size(min = 2) Float[] values,
                        @NonNull AnimationsValueChangeListener updateListener, int maxAnimationFps) {
    super(values, updateListener, maxAnimationFps);
  }

  @NonNull
  @Override
  TypeEvaluator provideEvaluator() {
    return new FloatEvaluator();
  }
}

File: platform/android/MapLibreAndroid/src/main/java/org/maplibre/android/location/MapLibreFloatAnimator.kt
package org.maplibre.android.location

import android.animation.FloatEvaluator
import android.animation.TypeEvaluator
import androidx.annotation.Size
import org.maplibre.android.location.MapLibreAnimator.AnimationsValueChangeListener

internal open class MapLibreFloatAnimator(
    @Size(min = 2) values: Array<Float>,
    updateListener: AnimationsValueChangeListener<Float>,
    maxAnimationFps: Int,
) : MapLibreAnimator<Float>(values, updateListener, maxAnimationFps) {
    override fun provideEvaluator(): TypeEvaluator<*> = FloatEvaluator()
}
